onnx/defs/nn/defs.cc

```diff
@@ -1542,4 +1534,73 @@ ONNX_OPERATOR_SET_SCHEMA(
         })
         .SetDoc(TfIdfVectorizer_ver9_doc));
 
+static const char* StringNormalizer_ver10_doc = R"DOC(
+StringNormalization performs string operations for basic cleaning. 
+This operator has only one input (denoted by X) and only one output 
+(denoted by Y). This operator first examines the elements in the X, 
+and remove elements specified in "stopwords" attribute. 
+Note that an implementation should sequentially remove "stopwords[0]," then "stopwords[1]," 
+and so on. After removing stop words, the intermediate result can be further lowercased, 
+uppercased, or just returned depending the "casechangeaction" attribute.
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    StringNormalizer,
+    10,
+    OpSchema()
+        .Input(0, "X", "Strings to normalize", "T")
+        .Output(0, "Y", "Normalized strings", "T")
+        .TypeConstraint(
+            "T",
+            {"tensor(string)"},
+            "Input/Output is a UTF-8 string tensor")
+        .Attr(
+            "casechangeaction",
+            "string enum that cases output to be lowercased/uppercases/unchanged. Valid values are \"LOWER\", \"UPPER\", \"NONE\"",
+            AttributeProto::STRING)
+        .Attr(
+            "is_case_sensitive",
+            "Boolean. Whether the identification of stop words in X is case-sensitive.",
+            AttributeProto::INT)
+        .Attr(
+            "stopwords",
+            "List of stop words. If not set, no work would be removed from X.",
+            AttributeProto::STRINGS,
+            OPTIONAL)
+        .Attr(
+            "locale",
+            "Environment dependent string that denotes the locale according to which output strings needs to be upper/lowercased."
+            "Default en_US or platform specific equivalent",
+            AttributeProto::STRING,
+            OPTIONAL)
+        .SetDoc(StringNormalizer_ver10_doc)
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          auto output_elem_type = ctx.getOutputType(0)->mutable_tensor_type();
+          output_elem_type->set_elem_type(TensorProto::STRING);
+          if (hasInputShape(ctx, 0)) {
```

Comment:
small comment: 

change to if(!hasInputShape(ctx, 0)) {...} to avoid big if block?